integration/integration.go

```diff
@@ -117,41 +117,6 @@ func findPodForJob(clientset *kubernetes.Clientset, name string, tout, timer tim
 						break podfailed
 					}
 
-					// Pod still working
```

Comment:
(GitHub won't let me put comments in code that hasn't changed, so sorry this isn't right next to the code I'm referring to.)

I don't think it's worth spending the time to change now, but there are two things I don't love about this:
- It's querying across all the pods in the namespace - why not just look at the job? 
- As a general rule, polling has a bad smell and it's much better to be event-driven. I think you probably needed [`JobInformer`](https://github.com/kubernetes/client-go/blob/master/informers/batch/v1/job.go)

Reply:
Lines 114-118 cover the case of the pod failing, so I don't think you have addressed #551 yet. Do you expect these jobs (and the pods within them) to ever fail except in cases where the whole integration test should fail?  

I would have thought that if a job fails, it should just fail the integration test. Or is there some condition under which we expect kube-bench to fail temporarily? If there is, at a minimum we should comment it and really we should be checking that the failure only happens in that condition.

Reply:
It turns out that the pods were failing because the control plane was not completely ready by the time the job is deployed. `kube-bench` was failing trying to find a way to get the `Kubernetes` version. This is because the API Server was not accepting requests and then tried to use `kubectl`, which is a symlink in `KIND` and failed to execute.  To fix #551,  I used a function provided by `KIND` to wait until the open control plane is "ready". One thing I noticed is that in most situations when `KIND` reports that the control plane is ready, the first pod deployed works, but sometimes, the first pod fails and the second pod success. This is why I have loop expecting failed pods and trying again. I'll give the [JobInformer](https://github.com/kubernetes/client-go/blob/master/informers/batch/v1/job.go) a try.

Reply:
I looked at the **`JobInformer`** and I believe using it would add complexity without providing much more value than just performing a poll a few times per job to find the corresponding.

1. As far as I can tell the **`JobInformer`** provides [Job](https://github.com/kubernetes/api/blob/7dc09db16fb8/batch/v1/types.go#L28) objects without providing the name of the actual pod that successfully executed **`kube-bench`**, and then a search for the pod would need to be performed.
2. In the case of performance improvement, the bulk of the time is taken by `KIND` brining the Kubernetes cluster up, about 2 minutes.
3. Finding the Pods the first time takes around 8-10 seconds, but finding the pods for subsequent jobs only takes a fraction of that.
4. I added a Label Selector to find the Pod so the Kubernetes API query would be job specific.

Reply:
I agree, as I said in my comment I don't think it's worth spending time to change to JobInformer at this point, just that it would have been a cleaner way to do it

Reply:
> It turns out that the pods were failing because the control plane was not completely ready by the time the job is deployed. kube-bench was failing trying to find a way to get the Kubernetes version. This is because the API Server was not accepting requests...

I don't understand this - you use the API from here to create the Job that runs the test, which starts a pod which starts kube-bench. So kube-bench can't be attempting API requests until some time after this test code has successfully made requests of the API Server, right?

Reply:
It's odd that from time to time, even after waiting for the `KIND` to indicate the control plane is up and running, the first Pod fails with the following:
```console
I0108 16:09:00.538760    1720 kubernetes_version.go:94] HTTP ERROR: Get https://kubernetes.default.svc/version: dial tcp: lookup kubernetes.default.svc on 10.96.0.10:53: no such host
W0108 16:09:00.546633    1720 util.go:315] 
Unable to find the programs kubectl or kubelet in the PATH.
These programs are used to determine which version of Kubernetes is running.
Make sure the /usr/bin directory is mapped to the container, 
either in the job.yaml file, or Docker command.

For job.yaml:
...
- name: usr-bin
  mountPath: /usr/bin
...

For docker command:
   docker -v $(which kubectl):/usr/bin/kubectl ....

Alternatively, you can specify the version with --version
   kube-bench --version <VERSION> ...

unable to determine benchmark version: Version check failed: unable to find the programs kubectl or kubelet in the PATH
Alternatively, you can specify the version with --version
```
Looks like `CoreDNS` is not ready or `DNS` completely setup for this Pod  by the time the code is attempting to access the Kubernetes API using `https://kubernetes.default.svc/version`

I have an idea on how to fix that, but I think it'll be better if I create another issue just for that so that we finish with this PR.

Reply:
You could do that under #551